develop/tutorials/articles/260-forms/01-forms-storage-adapters.markdown

```diff
@@ -71,13 +73,15 @@ the storage adapter registry. Next override the `doCreateMethod` to return a
         return fileId;
     }
 
-In addition to returning the file ID, it's necessary to add a storage link via
-the `DDMStorageLinkLocalService`. WHY DO WE NEED TO ADD A RECORD TO THE
-DDMSTORAGELINK TABLE? Pass the method the class name ID as retrieved by
-`PortalUtil.getClassNameId`, the `fileId` as the pseudo-primary key for the
-storage type, the structure version ID, and the service context. There's also a
-call to a `saveFile` method, which serializes the forms record's values and uses
-two additional utility methods to write a `File`:
+In addition to returning the file ID, add a storage link via the
+`DDMStorageLinkLocalService`. The DDM Storage Link is used to associate each
+form record with the form it's being entered for.
```

Comment:
Actually, according to the database tables, the`DDMStorageLink` creates a link between the form's record and the `DDMStructure` associated to the Form. Please check the database table `DDMStorageLink`. 😉

Reply:
Thanks. So how does DDMStructure relate to DDMFormInstance? I've been curious about how these tables and their services work together.

Reply:
Hey @rbohl, from what I can see, the `DDMStorageLink` relates to the Forms' record instead of the form, because the storage entity is more related to the record than the form, once it describes the way the records will be stored.

Additionally, the `DDMStructure` / `DDMStorageLink` is used by other entities of the portal, like Web Content and Dynamic Data Lists, hence the `DDMStructure` / `DDMStorageLink` needs to be independent from a specific entity.